HollowKnightComponent.cs

```diff
@@ -185,59 +193,71 @@ public class HollowKnightComponent {
                     }
                 }
                 else if (!((gameState == GameState.INACTIVE && uIState == UIState.INACTIVE) || (gameState == GameState.MAIN_MENU))) {
-                    if (CheckSplit(split, nextScene, sceneName)) {
-                        splitsDone.Add(split);
-                        lastSplitDone = split;
+                    if (CheckSplit(Split, nextScene, sceneName) == SplitterAction.Split) {
+                        splitsDone.Add(Split);
+                        lastSplitDone = Split;
                         menuSplitHelper = false;
-                        if (hasLog || !Console.IsOutputRedirected) WriteLogWithTime("Split: " + split);
-                        return true;
+                        if (hasLog || !Console.IsOutputRedirected) WriteLogWithTime("Split: " + Split);
+                        return SplitterAction.Split;
                     }
                 }
             }
-            return false;
+            return SplitterAction.Pass;
         }
 
-        private bool OrderedSplits(GameState gameState, UIState uIState, string nextScene, string sceneName) {
-            SplitName split = settings.Splits[currentSplit];
+        private SplitterAction OrderedSplits(GameState gameState, UIState uIState, string nextScene, string sceneName) {
+            SplitName Split = settings.Splits[currentSplit];
 
-            if (split.ToString().StartsWith("Menu")) {
-                if (!menuSplitHelper) menuSplitHelper = split == SplitName.Menu || CheckSplit(split, nextScene, sceneName);
+            if (Split.ToString().StartsWith("Menu")) {
+                if (!menuSplitHelper) menuSplitHelper = Split == SplitName.Menu || CheckSplit(Split, nextScene, sceneName) != SplitterAction.Pass;
                 if (menuSplitHelper) {
-                    if (CheckSplit(SplitName.Menu, nextScene, sceneName)) {
+                    if (CheckSplit(SplitName.Menu, nextScene, sceneName) == SplitterAction.Split) {
+                        menuSplitHelper = false;
+                        if (hasLog || !Console.IsOutputRedirected) WriteLogWithTime("Split: " + Split);
+                        return SplitterAction.Split;
+                    } else if (CheckSplit(SplitName.Menu, nextScene, sceneName) == SplitterAction.Skip) {
                         menuSplitHelper = false;
-                        if (hasLog || !Console.IsOutputRedirected) WriteLogWithTime("Split: " + split);
-                        return true;
+                        if (hasLog || !Console.IsOutputRedirected) WriteLogWithTime("Skip: " + Split);
+                        return SplitterAction.Skip;
                     }
                 }
             } else {
-                if (CheckSplit(split, nextScene, sceneName)
+                if (CheckSplit(Split, nextScene, sceneName) == SplitterAction.Split
+                        && !((gameState == GameState.INACTIVE && uIState == UIState.INACTIVE) || (gameState == GameState.MAIN_MENU))) {

```

Comment:
instead of copy/pasting this chunk, we could factor it out into something like `gameInactiveOrMainMenu`